crates/wysiwyg/src/composer_model/hyperlinks.rs

```diff
@@ -87,6 +91,32 @@ where
         self.create_update_replace_all()
     }
 
+    fn add_http_scheme<'a>(&'a mut self, link: &'a S) -> S {
+        let mut new_link = link.clone();
+        let string = link.to_string();
+        let str = string.as_str();
+
+        match Url::parse(str) {
+            Ok(url) => url,
+            Err(ParseError::RelativeUrlWithoutBase) => {
+                let is_email = EmailAddress::is_valid(str);
+
+                if is_email {
+                    new_link.insert(0, &S::from("mailto:"));
+                } else {
+                    new_link.insert(0, &S::from("https://"));
+                }
+                return new_link;
+            }
+            Err(_) => {
+                return new_link;
```

Comment:
I don't think we really need to return something here, we will return the new_link in any cases at the end of the method anyway. 
It's fine to write something like `Ok(_) => {}` for the standard working case and have the match return nothing.